src/init.cpp

```diff
@@ -1413,7 +1420,7 @@ bool AppInitMain()
 
     bool fLoaded = false;
     while (!fLoaded && !ShutdownRequested()) {
-        bool fReset = fReindex;
+        bool fReset = fReindex || fReloadxfield;
```

Comment:
fReset flag resets the blocks db, coins db and created a flag called `reindexing` in level db. In case the node restarts when reindexing is in progress this flag helps it to identify that reindexing didn't complete. We need the same path to be taken for both "-reindex" and "-reloadxfield". So we need fReset for reloadxfield too.

Reply:
You mean that if a node is restarted during the reloadxfield process, is it to determine that the processing has not completed?

In this case(without reindex option but with reloadxfield option), reindexing is not performed even if this flag is set?

Reply:
> You mean that if a node is restarted during the reloadxfield process, is it to determine that the processing has not completed?
>
Actually, fReindex is the leveldb flag used to determine whether reindex completed or not. fReset is set based on the command line option as a backup so that later on in the code when fReindex may be different we use fReset

> In this case(without reindex option but with reloadxfield option), reindexing is not performed even if this flag is set?

`fReset` is set so that reindexing is performed in both cases i.e `reloadxfield` and `reindex`